src/Application/ApplicationFileProcessor.php

```diff
@@ -108,21 +121,29 @@ public function run(Configuration $configuration, InputInterface $input): Proces
 
     /**
      * @param string[] $filePaths
+     * @param string[] $allFiles
      * @param callable(string $file): void|null $preFileCallback
      * @param callable(int $fileCount): void|null $postFileCallback
      */
     public function processFiles(
         array $filePaths,
         Configuration $configuration,
+        ?array $allFiles = null,
         ?callable $preFileCallback = null,
-        ?callable $postFileCallback = null
+        ?callable $postFileCallback = null,
     ): ProcessResult {
         /** @var SystemError[] $systemErrors */
         $systemErrors = [];
 
         /** @var FileDiff[] $fileDiffs */
         $fileDiffs = [];
 
+        if ($allFiles === null) {
```

Comment:
If we are running in parallel, this list will be null, so we load it from the cache